app/services/workflow_executions/cancel_service.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module WorkflowExecutions
+  # Service used to initiate the cancelation of a WorkflowExecution
+  class CancelService < BaseService
+    def initialize(workflow_execution, user = nil)
+      super(user, {})
+      @workflow_execution = workflow_execution
+    end
+
+    def execute
+      return false unless @workflow_execution.cancellable?
+
+      @workflow_execution.state = 'canceling'
+      @workflow_execution.save
```

Comment:
I believe this can be removed since you are doing the `return unless @workflow_execution.save` below

Reply:
Makes sense, removed in 4db95a140ab67bf415e108178f3de20571e8834c